### PR TITLE
fix: make rekor.pub a variable

### PR DESCRIPTION
**Pull request by laurentsimon**
Signed-off-by: laurentsimon <laurentsimon@google.com>

closes https://github.com/sigstore/sigstore-js/issues/215

#### Summary
The rekor.pub key is stored as a file. The file path does not resolve properly if the caller calls the API (vs CLI). When using [ncc](https://github.com/vercel/ncc), the file is not packaged with the final .js file. Using a variable solves these problems. Another option seems to use `require.resolve()`; which is the fix we agreed on. 

This is probably a temporary fix until TUF is used.

### CODE REVIEW

File: src/tlog/keys.ts [hunk 1]
```diff
@@ -14,18 +14,18 @@ See the License for the specific language governing permissions and
...
   // TODO: This should be be loaded via TUF
-  const pem = fs.readFileSync(
-    path.resolve(__dirname, '../../store/rekor.pub'),
-    'utf-8'
-  );
+  const pem = `
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
+kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
+-----END PUBLIC KEY-----`
```
Comment (feelepxyz) [inline src/tlog/keys.ts:27]: This file should in theory be relocated by `ncc` > [webpack-asset-relocator-loader](https://github.com/vercel/webpack-asset-relocator-loader) but I had a go at this and looks like it doesn't like the typescript compiled code here as it's trying to pattern match on `path.*` and TS generates this: `path_1.default.resolve`.

What does seem to work is changing this to use `require.resolve`:

```suggestion
  const pem = fs.readFileSync(
    require.resolve('../../store/rekor.pub'),
    'utf-8'
  );
```

File: src/tlog/keys.ts [hunk 2]
```diff
@@ -14,18 +14,18 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 import { createPublicKey, KeyObject } from 'crypto';
```
Comment (feelepxyz) [inline src/tlog/keys.ts:16]:
```suggestion
import { createPublicKey, KeyObject } from 'crypto';
import fs from 'fs';
```

**Comment by laurentsimon** [comment]
Thanks @feelepxyz! I've accepted your changes.

**Review by feelepxyz — APPROVED** [review]
Appreciate the fix!

**Review by bdehamer — CHANGES_REQUESTED** [review]
I think the `rekor.pub` file needs to be added back in now.

**Comment by laurentsimon** [comment]
I have added the rekor.pub file back in. PTAL

**Review by feelepxyz — APPROVED** [review]

**Review by bdehamer — APPROVED** [review]
👍

**Pull request merged**